packages/tldraw/src/lib/ui/context/actions.tsx

```diff
@@ -218,13 +289,17 @@ export function ActionsProvider({ overrides, children }: ActionsProviderProps) {
 					trackEvent('export-as', { format: 'json', source })
 					exportAs(ids, 'json', getExportName(editor, defaultDocumentName))
 				},
+				enabled() {
+					return pageHasShapes
```

Comment:
so, i would have expected maybe in the `onSelect` function that we could get rid of the extra logic. i think with these changes that `onSelect` should correctly assume that it won't be called _unless_ `enabled()` returns true.

otherwise, it feels like we're duplicating a lot of logic can't really take advantage of this new paradigm. wdyt?

Reply:
Was wondering the same thing. That said the end users can use the actions without checking if it is enabled or not, so I guess it's better if we are a bit more defensive about applying these actions. Maybe `onSelect` could somehow call `enabled` though.

Reply:
i hesitate to say this...but should these be converted into classes (from being an object) so that they can call `enabled()`? 😬 

at the very least, even if we're not calling `enabled()` directly, can `onSelect()` just at least do:
`if (!pageHasShapes) return`